installer/DetailsHandler.php

```diff
@@ -126,28 +126,34 @@ public function getAdminEmployeeLastName()
      * Get the organization email address
      * @return mixed
      */
-    public function getOrganizationEmailAddress()
+    public function getOrganizationEmailAddress($invalid = false)
     {
         $configurationDataSet = $this->getCongigIni();
 
-        if ($_SESSION['defUser']['organizationEmailAddress']) {
-            return $_SESSION['defUser']['organizationEmailAddress'];
+        $email = $_SESSION['defUser']['organizationEmailAddress'] ? $_SESSION['defUser']['organizationEmailAddress'] :$configurationDataSet["organizationEmailAddress"];
+
+        if ($invalid) {
+            $email = $this->takeUserInput("Invalid Email Address. Please enter Email Address:");
         }
-        return $configurationDataSet["organizationEmailAddress"];
+
+        return $this->isValidUserInput("Organization Email Address ", $email) ? $email : $this->getOrganizationEmailAddress(true);
     }
 
     /**
      * Get the Admin employee contact number
-     * @return mixed
+     * @param bool $invalid
+     * @return mixed|string
      */
-    public function getContactNumber()
+    public function getContactNumber($invalid = false)
     {
         $configurationDataSet = $this->getCongigIni();
+        $contactNumber = $_SESSION['defUser']['contactNumber'] ? $_SESSION['defUser']['contactNumber'] : $configurationDataSet["contactNumber"];
 
-        if ($_SESSION['defUser']['contactNumber']) {
-            return $_SESSION['defUser']['contactNumber'];
+        if ($invalid) {
```

Comment:
use something like empty($this->error['contact-number'])

Reply:
updated